locator/token_metadata.cc

```diff
@@ -534,16 +535,17 @@ class token_metadata_impl final {
     }
 
 #endif
+    // Returns nodes that are officially part of the ring. It does not include
+    // node that is still joining the cluster, e.g., a node that is still
+    // streaming data before it finishes the bootstrap process and turns into
+    // NORMAL status.
```

Comment:
Nitpick: maybe the name of the function should be changed to get_all_normal_endpoints() to emphesize this?
But maybe it's not worth the churn to the using code.

Reply:
Kamil and I discussed the renaming. We agreed to do cosmetic changes in follow up patch.

Reply:
> @asias the function gossiper::get_up_endpoint_count() also needs to look at the state of endpoints, but does this more efficiently (doesn't need to look at each node 256 times). Couldn't we do the same here, and check for NORMAL nodes.

No. token_metada is more authoritative source of the ring status. We'd better not depend on gossip for the ring status.